src/voter/mod.rs

```diff
@@ -420,6 +430,102 @@ fn instantiate_last_round<H, N, E: Environment<H, N>>(
 	}
 }
 
+/// Trait for querying the state of the voter. Used by `Voter` to return a queryable object without exposing too may
+/// data types.
+pub trait VoterState<Id: Eq + std::hash::Hash> {
+	/// Returns a plain data type, `report::VoterState`, describing the current state of the voter relevant to the
+	/// voting process.
+	fn voter_state(&self) -> report::VoterState<Id>;
```

Comment:
Maybe we could rename this to `get` otherwise currently it looks like this: `voter.voter_state().voter_state()`.

Reply:
Updated!